app/services/metrics/for_key_metric_dashboard_service.rb

```diff
@@ -100,7 +100,7 @@ def validations
         top_validation_screens(time_period).map do |screen_and_count|
           {
             checks: screen_and_count[1],
-            screen: screen_and_count[0],
+            screen: Flow::Handler.url_fragment(screen_and_count[0]),
```

Comment:
Looks good. What I don't quite understand is how the metrics will work here. So it looks like the `screen` value will now be the new url fragment. How will the existing data under the old name and the new data under the new URL be consolidated?

Reply:
Ah yes, good question, and I can see that this looks potentially problematic. The reason this is actually fine is that we are _replacing_ the complete Geckoboard dataset every time this service is called (at the top of the file we do `..._dataset.put` instead of `..._dataset.post` - a `put` means "this is the new dataset" rather than "add this to the dataset"). So the first time this runs in production, all references on Geckoboard to old step names will immediately disappear and be replaced by new step URL fragments.

Reply:
Ah great! Just my misunderstanding with how the Geckoboard dataset gets created :)